packages/starknet-snap/src/utils/starknetUtils.ts

```diff
@@ -739,6 +748,65 @@ export const getAccContractAddressAndCallDataLegacy = (publicKey) => {
   };
 };
 
+/**
+ * Creates a new account on the specified network and records deployment information to the state.
+ *
+ * This function generates a new account address and call data based on the provided public key,
+ * deploys the account on the network, and records the deployment details such as contract address
+ * and transaction hash into the state using the provided `recordAccountDeployment` function.
+ *
+ * Optionally, if `waitMode` is enabled, the function will wait for the transaction to be confirmed
+ * before returning the account address and transaction hash.
+ *
+ * @param network - The network on which to deploy the account.
+ * @param publicKey - The public key used to derive the account address.
+ * @param privateKey - The private key used to sign the deployment transaction.
+ * @param cairoVersion - The version of Cairo to use for the account deployment.
+ * @param recordAccountDeployment - A function to record deployment information into the state.
+ * @param waitMode - If true, waits for the transaction to be confirmed before returning. Defaults to false.
+ * @returns An object containing the deployed account address and the transaction hash.
+ */
+export const createAccount = async (
+  network: Network,
+  publicKey: string,
+  privateKey: string,
+  cairoVersion: CairoVersion,
+  recordAccountDeployment: RecordAccountDeploymentFn,
+  waitMode = false,
+) => {
+  const { address, callData } = getAccContractAddressAndCallData(publicKey);
+  // Deploy account will auto estimate the fee from the network if not provided
+  const deployResp = await deployAccount(
```

Comment:
```suggestion
  const {
    contract_address: contractAddress,
    transaction_hash: transactionHash,
  } = await deployAccount(
```

let de structure it and have better naming to avoid var pattern "xxx_xxx" every where